src/topology/noetherian_space.lean

```diff
@@ -157,14 +158,20 @@ lemma noetherian_space.discrete [noetherian_space α] [t2_space α] : discrete_t
 local attribute [instance] noetherian_space.discrete
 
 /-- Spaces that are both Noetherian and Hausdorff is finite. -/
-@[priority 100]
-noncomputable
-def noetherian_space.fintype [noetherian_space α] [t2_space α] : fintype α :=
-set.fintype_of_finite_univ (noetherian_space.is_compact set.univ).finite_of_discrete
+lemma noetherian_space.finite [noetherian_space α] [t2_space α] : finite α :=
+begin
+  letI : fintype α :=
+    set.fintype_of_finite_univ (noetherian_space.is_compact set.univ).finite_of_discrete,
+  apply_instance
+end
 
 @[priority 100]
-instance noetherian_space.of_fintype [fintype α] : noetherian_space α :=
-by { classical, exact ⟨@@fintype.well_founded_of_trans_of_irrefl (subtype.fintype _) _ _ _⟩ }
+instance noetherian_space.of_fintype [finite α] : noetherian_space α :=
+begin
+  casesI nonempty_fintype α,
+  classical,
+  exact ⟨@@fintype.well_founded_of_trans_of_irrefl (subtype.fintype _) _ _ _⟩
+end
```

Comment:
We should replace `fintype.well_founded_of_trans_of_irrefl` by `finite.well_founded_of_trans_of_irrefl`.